### PR TITLE
Refactor copy of slices and maps

### PR DESCRIPTION
**Pull request by Honny1**
This PR refactors:

- Operations with slices:
   - Removes duplicate copy*Slice functions using a generic copy function or replaces them with the `slices.Clone` function. 
   - Also simplifies the `stringSliceWithoutValue` function. 
   - **These changes should not change the behavior.**
- Copying maps:
   - Removes the duplicate copy*Map function using the general function `newMapFrom`. 
   - Reduces the allocation of empty maps using the `copyMapPrefferingNil` function. 
   - **This change may affect the behavior so that instead of an empty allocated map, a `nil` will be returned.**

Follows up on: #2122

### CODE REVIEW

File: store.go [hunk 1]
```diff
@@ -1783,13 +1759,13 @@ func (s *store) CreateContainer(id string, names []string, image, layer, metadat
...
-		options.LabelOpts = copyStringSlice(cOptions.LabelOpts)
+		options.IDMappingOptions.UIDMap = copySlicePreferringNil(cOptions.IDMappingOptions.UIDMap)
+		options.IDMappingOptions.GIDMap = copySlicePreferringNil(cOptions.IDMappingOptions.GIDMap)
+		options.LabelOpts = copySlicePreferringNil(cOptions.LabelOpts)
 		options.Flags = maps.Clone(cOptions.Flags)
-		options.MountOpts = copyStringSlice(cOptions.MountOpts)
+		options.MountOpts = copySlicePreferringNil(cOptions.MountOpts)
 		options.StorageOpt = copyStringStringMap(cOptions.StorageOpt)
-		options.BigData = copyContainerBigDataOptionSlice(cOptions.BigData)
+		options.BigData = slices.Clone(cOptions.BigData)
```
Comment (kolyshkin) [inline store.go]: `slices.Clone` makes a shallow clone, i.e. `Data` slice is not really copied with this change.

**Review by kolyshkin — CHANGES_REQUESTED** [review]

**Comment by rhatdan** [comment]
/approved
LGTM

File: store.go (hunk 1 above)
Comment (Honny1) [reply store.go]: The previous `copyContainerBigDataOptionSlice` function used the `slices.Clone` function to copy data. So I would expect similar behavior.

Comment (nalind) [reply store.go]: It used `slices.Clone()` to copy the `Data` field in each element in the slice.  The godoc for `slices.Clone()` notes that it only copies using assignment, so a new slice would not be created for `Data` fields.

**Comment by rhatdan** [comment]
/approve
@kolyshkin @nalind @mtrmac @Luap99 @flouthoc @giuseppe PTAL

**Comment by nalind** [comment]
LGTM

**Pull request merged**